src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseConcreteTypeAnalyzer.cs

```diff
@@ -0,0 +1,232 @@
+                context.RegisterOperationAction(context => coll.HandleFieldInitializer((IFieldInitializerOperation)context.Operation), OperationKind.FieldInitializer);
+                context.RegisterOperationAction(context => coll.HandleVariableDeclarator((IVariableDeclaratorOperation)context.Operation), OperationKind.VariableDeclarator);
+                context.RegisterOperationAction(context => coll.HandleDeclarationExpression((IDeclarationExpressionOperation)context.Operation), OperationKind.DeclarationExpression);
+                context.RegisterOperationAction(context => coll.HandleReturn((IReturnOperation)context.Operation), OperationKind.Return);
+
+                context.RegisterSymbolEndAction(context =>
+                {
+                    Report(context, coll);
+                    coll.Dispose();
+                });
+            }, SymbolKind.NamedType);
+        }
+
+        /// <summary>
+        /// Given all the accumulated analysis state, generate the diagnostics.
+        /// </summary>
+        private static void Report(SymbolAnalysisContext context, Collector coll)
+        {
+            foreach (var field in coll.VirtualDispatchFields.Keys)
+            {
+                if (coll.FieldAssignments.TryGetValue(field, out var assignments))
+                {
+                    Report(field, field.Type, assignments, UseConcreteTypeForField);
+                }
+            }
+
+            foreach (var local in coll.VirtualDispatchLocals.Keys)
+            {
+                if (coll.LocalAssignments.TryGetValue(local, out var assignments))
+                {
+                    Report(local, local.Type, assignments, UseConcreteTypeForLocal);
+                }
+            }
+
+            foreach (var parameter in coll.VirtualDispatchParameters.Keys)
+            {
+                if (coll.ParameterAssignments.TryGetValue(parameter, out var assignments))
+                {
+                    if (parameter.ContainingSymbol is IMethodSymbol method)
+                    {
+                        if (CanUpgrade(method))
+                        {
+                            Report(parameter, parameter.Type, assignments, UseConcreteTypeForParameter);
+                        }
+                    }
+                }
+            }
+
+            foreach (var pair in coll.MethodReturns)
+            {
+                var method = pair.Key;
+                var returns = pair.Value;
+
+                if (CanUpgrade(method))
+                {
+                    Report(method, method.ReturnType, returns, UseConcreteTypeForMethodReturn);
+                }
+            }
+
+            void Report(ISymbol sym, ITypeSymbol fromType, ConcurrentBag<ITypeSymbol> assignments, DiagnosticDescriptor desc)
+            {
+                using var types = PooledHashSet<ITypeSymbol>.GetInstance(SymbolEqualityComparer.Default);
+                types.UnionWith(assignments);
```

Comment:
nit: I think this should functionally be the same.

```suggestion
                using var types = PooledHashSet<ITypeSymbol>.GetInstance(assignments, SymbolEqualityComparer.Default);
```

Reply:
Done.